src/components/Layout/Header/NavBar/NavBar.tsx

```diff
@@ -14,26 +17,52 @@ type NavBarProps = {
 export const NavBar = ({ isCompact, ...rest }: NavBarProps) => {
   const translate = useTranslate()
   const { routes: pluginRoutes } = usePlugins()
+  const groupColor = useColorModeValue('gray.300', 'gray.600')
+
+  const groupArray = useMemo(() => {
+    const allRoutes = union(routes, pluginRoutes).filter(route => !route.disable && !route.hide)
+    const groups = allRoutes.reduce(
+      (entryMap, e) => entryMap.set(e.category, [...(entryMap.get(e.category) || []), e]),
+      new Map(),
+    )
+    return Array.from(groups.entries())
+  }, [pluginRoutes])
 
   return (
-    <Stack width='full' flex='1 1 0%' {...rest}>
-      {union(routes, pluginRoutes)
-        .filter(route => !route.disable && !route.hide)
-        .map((item, idx) => {
-          return (
-            <MainNavLink
-              isCompact={isCompact}
-              as={ReactRouterLink}
-              key={idx}
-              leftIcon={item.icon}
-              href={item.path}
-              to={item.path}
-              label={translate(item.label)}
-              aria-label={translate(item.label)}
-              data-test={`navigation-${item.label.split('.')[1]}-button`}
-            />
-          )
-        })}
+    <Stack width='full' flex='1 1 0%' spacing={6} {...rest}>
+      {groupArray.map((group, id) => {
+        const name = group[0]
+        const values = group[1]
```

Comment:
Ideally would prefer this to be key-based but if groups are going to be arrays, let's make it:

```suggestion
        const [name, values] = group
```

Or even better, destructure it on the line above